src/state/models/cache/handle-resolutions.ts

```diff
@@ -0,0 +1,5 @@
+import {LRUMap} from 'lru_map'
+
+export class HandleResolutionsCache {
```

Comment:
`lru_map` is a good solution for this but I don't think we would want this to be clogging up the JS thread. Not sure about the performance implications but it is something to be wary of (re: memory issues). Could use https://reactnative.dev/docs/asyncstorage for async storage or https://github.com/mrousavy/react-native-mmkv for sync storage

Reply:
I will caveat this and say its _probably_ fine

Reply:
Wouldnt the context switching of messaging the main process be worse than the processing overhead of the LRU manager?

Reply:
I don't think I know the internals of react native to give an answer I am confident about with this matter.

But one more thing to consider is that the `lru_map` will be wiped out on every app restart

Reply:
Yeah for now I'm good with that, we dont want to persist that long